extensions/smallrye-reactive-messaging-kafka/deployment/src/main/java/io/quarkus/smallrye/reactivemessaging/kafka/deployment/SmallRyeReactiveMessagingKafkaProcessor.java

```diff
@@ -46,18 +49,34 @@ public void build(BuildProducer<ReflectiveClassBuildItem> reflectiveClass) {
                         .build());
     }
 
+    /**
+     * Handles the serializer/deserializer detection and whether or not the graceful shutdown should be used in dev mode.
+     */
     @BuildStep
-    public void defaultSerdeConfig(ReactiveMessagingKafkaBuildTimeConfig buildTimeConfig,
+    public void defaultChannelConfiguration(
+            LaunchModeBuildItem launchMode,
+            ReactiveMessagingKafkaBuildTimeConfig buildTimeConfig,
+            ReactiveMessagingKafkaConfig runtimeConfig,
             CombinedIndexBuildItem combinedIndex,
             BuildProducer<RunTimeConfigurationDefaultBuildItem> defaultConfigProducer) {
 
-        if (!buildTimeConfig.serializerAutodetectionEnabled) {
-            return;
-        }
-
         DefaultSerdeDiscoveryState discoveryState = new DefaultSerdeDiscoveryState(combinedIndex.getIndex());
+        if (buildTimeConfig.serializerAutodetectionEnabled) {
+            discoverDefaultSerdeConfig(discoveryState, defaultConfigProducer);
+        }
 
-        discoverDefaultSerdeConfig(discoveryState, defaultConfigProducer);
+        if (launchMode.getLaunchMode() == LaunchMode.DEVELOPMENT) {
```

Comment:
What about test mode as well? Will this potentially add extra time at the end of test runs?

Reply:
Good idea. Should it be controlled by the same configuration property? 

Right now it uses `enableGracefulShutdownInDevMode`, `enableGracefulShutdownInDevAndTestMode` is a bit long, but with autocompletion, it should not be a problem.

Reply:
Added.